detox/src/ipc/ipc.test.js

```diff
@@ -116,69 +186,108 @@ describe('IPC', () => {
         await ipcClient2.init();
       });
 
-      it('should report test results', async () => {
-        const testResults1 = [
-          {
+      describe('reportTestResults', () => {
+        it('should report test results and synchronize test results', async () => {
+          const testResults = [
+            {
+              testFilePath: 'file1',
+              success: true,
+            },
+          ];
+
+          await ipcClient1.reportTestResults(testResults);
+          expect(ipcServer.sessionState.testResults).toEqual(testResults);
+          expect(ipcClient1.sessionState.testResults).toEqual(testResults);
+
+          await sleep(10); // broadcasting happens with a delay
```

Comment:
Adding a "broadcasting" sleep because it happens asynchronously, later in the macro-tasks queue.